### ISSUE TITLE
SC3012: warning does not escape `<`/`>` in `test`

### ISSUE THREAD

**Issue by e-kwsm** [koalaman/shellcheck#3172]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC3012
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
[ "$1" '<' b ]
test "$2" \> b
```

#### Here's what shellcheck currently says:
```
Line 2	SC3012: In POSIX sh, lexicographical \< is undefined.
Line 3	SC3012: In POSIX sh, lexicographical > is undefined.
```
#### Here's what I wanted or expected to see:

As shown above, `<` and `>` in `[` and `test` need escape or quotation to avoid misinterpretation as redirection, therefore the warning must be

```
Line 3	SC3012: In POSIX sh, lexicographical \> is undefined.
```
